app/models/spree/gateway/mollie_gateway.rb

```diff
@@ -215,23 +222,36 @@ def update_payment_status(payment)
           api_key: get_preference(:api_key)
       )
 
-      MollieLogger.debug("Updating order state for payment. Payment has state #{mollie_payment.status}")
-
+      MollieLogger.debug("Checking Mollie payment status. Mollie payment has status #{mollie_payment.status}")
       update_by_mollie_status!(mollie_payment, payment)
     end
 
     def update_by_mollie_status!(mollie_payment, payment)
       case mollie_payment.status
-        when 'paid'
-          payment.complete! unless payment.completed?
+      when 'paid'
+        # If Mollie payment is already paid and refunded amount is more than 0, don't update payment
+        if mollie_payment.paid? && mollie_payment.amount_refunded.value > 0
+          MollieLogger.debug('Payment is refunded. Not updating the payment status within Spree.')
+          return
+        end
+
+        if payment.completed?
+          MollieLogger.debug('Payment is already completed. Not updating the payment status within Spree.')
+        end
+
+        # If order is already paid for, don't mark it as complete again.
+        unless payment.completed?
+          payment.complete!
           payment.order.finalize!
```

Comment:
- `payment.complete!` will transition the payment to the complete state via state machine.
- `payment.order.finalize!` will complete the order and send the order confirmation email. This should only be done once.